README.md

```diff
@@ -1,207 +1,304 @@
-```GPdatabase```     |   gp.GPDatabase: The GreenplumDB connection instance 
-```output_meta```     |    gp.GPTableMeta:  The output metadata (i.e. the details of output table). The output_meta.signature must be set. Please see below for the details of gp.GPTableMeta. If the output_meta.name is `None`, function will return a pd.dataframe as the results instead of store the results in a new table with name output_mate.name.
-```clear_existing```     |      Bool: Whether clear existing output table stored in GreenplumDB before executing the query 
-```runtime_id```     |      Used by "plcontainer" runtime type only. The runtime id is set by plcontainer to specify a runtime cnofiguration. See plcontainer for more information. The argument type is string, e.g. "plc_python_shared" 
-```runtime_type```     |      value should be "plcontainer" or "plpythonu" 
-```**kwargs```     |       if the function py_func contains extra arguments other than input table columns, you can append them after all of the required arguments list.
+Finally when we are done, we can save the resulting table to the database, either temporarily or persistently:
 
+```python
+t_n.save_as(table_name="t_n", temp=True)
+```
 
-### Return Value
+# `JOIN`-ing Two Tables
 
- A `pd.dataframe` that contains the result if the output_meta.name is set. Otherwise, it returns a `None`.
+We can also `JOIN` two tables with GreenplumPython. For example, suppose we have two tables like this:
 
+```python
+rows = [(1, "'a'",), (2, "'b'",), (3, "'c'",), (4, "'d'")]
+t1 = gp.values(rows, db=db, column_names=["id, val"])
+display(t1)
+```
 
-## Utils Functions Usage
+<table>
+<thead>
+<tr><th style="text-align: right;">  id</th><th>val  </th></tr>
+</thead>
+<tbody>
+<tr><td style="text-align: right;">   1</td><td>a    </td></tr>
+<tr><td style="text-align: right;">   2</td><td>b    </td></tr>
+<tr><td style="text-align: right;">   3</td><td>c    </td></tr>
+<tr><td style="text-align: right;">   4</td><td>d    </td></tr>
+</tbody>
+</table>
 
-```Python
-import GreenplumPython as gp
-import Pandas as pd
+```python
+rows = [(1, "'a'",), (2, "'b'",), (3, "'a'",), (4, "'b'")]
+t2 = gp.values(rows, db=db, column_names=["id, val"])
+display(t2)
+```
 
-# init connection instance
-connection = gp.Connection()
+<table>
+<thead>
+<tr><th style="text-align: right;">  id</th><th>val  </th></tr>
+</thead>
+<tbody>
+<tr><td style="text-align: right;">   1</td><td>a    </td></tr>
+<tr><td style="text-align: right;">   2</td><td>b    </td></tr>
+<tr><td style="text-align: right;">   3</td><td>a    </td></tr>
+<tr><td style="text-align: right;">   4</td><td>b    </td></tr>
+</tbody>
+</table>
 
-# connect to GPDB
-conn = gp.connection.connect(dbIP, dbPort, dbname, username, password, ...)
+We can `JOIN` the two table like this:
 
-conn1 = gp.connection.connect(dbIP, dbPort, dbname, username, password, ...)
+```python
+t_join = t1.join(
```

Comment:
Since we have different functions for different types of JOIN, will it be better to specify one type here and to list all other possibilities?